repository/gogit_testing.go

```diff
@@ -62,3 +75,18 @@ func SetupGoGitReposAndRemote() (repoA, repoB, remote TestedRepo) {
 
 	return repoA, repoB, remote
 }
+
+func RepoDir(t *testing.T, repo TestedRepo) string {
```

Comment:
Hmmm, I'm not sure about this.

My first concern is that if you really need to access the repo path for testing, it might be better to put an extra function in [repoTest](https://github.com/MichaelMure/git-bug/blob/44f61b66775e77fd8b42ba3ea57b2c323463b725/repository/repo.go#L205).

But then ... I'm not sure it's correct to test actual files, or even assume there *is* files. For example, `mockRepo` is purely in memory. In the future, we could have other alternate implementation for a Repo: in the LocalStorage of a browser, in a database ... `Repo` is a very good interface to transplant git-bug onto another system.

Which means ... [TestGoGitRepo_Indexes](https://github.com/MichaelMure/git-bug/blob/f3d316d16cc9fadd90faaaf77becbb82f1e09367/repository/gogit_test.go#L63) should likely be implemented as a reusable `func RepoBleveTest(t *testing.T, repo RepoBleve)` in `repo_testing.go`, and ignore files entirely. If the indexing works, do we really care about the files?

Reply:
> The TestedRepo itself includes a function called EraseFromDisk so to some degree, we're expecting that the repo used for testing is disk backed.

Not really, at least in my head. `TestedRepo` is a low quality catch-all to add testing specific capabilities. If you look at the implementation of `mockRepoTest`, `EraseFromDisk` just do nothing. Admittedly, the naming could be improved, or even better this should be removed entirely as it's now possible with this PR.

1. I see that one as an extra test, specific for `GoGitRepo`. In this case it's fine to bypass the interface boundary and test internal specific things that are specially error prone (where the files end up in different scenario).

2. A similar case could be made about this one the need for a generic test for the index remains.